lib/spack/spack/stage.py

```diff
@@ -45,18 +75,16 @@ def _first_accessible_path(paths):
                 if can_access(path):
                     return path
             else:
-                # The path doesn't exist so create it and adjust permissions
-                # and group as needed (e.g., shared ``$tempdir``).
-                prefix = os.path.sep
-                parts = path.strip(os.path.sep).split(os.path.sep)
-                for part in parts:
-                    prefix = os.path.join(prefix, part)
-                    if not os.path.exists(prefix):
-                        break
-                parent = os.path.dirname(prefix)
-                gid = os.stat(parent).st_gid
-                mkdirp(path, group=gid, default_perms='parents')
-
+                # Remove adjacent occurance of `$user` (e.g., `$tempdir/$user`
+                # where `$tempdir` includes `$user`).
```

Comment:
That sounds good: I think this is more cleanly done by just checking if `$tempdir` contains a directory that is equal to `$user` rather than examining the full path we construct; that logic could be isolated into a separate function.

Reply:
That does interfere with my (informal) goal of ultimately moving all test directories to be created under a single `$tempdir` directory to facilitate cleanup. :)

I'm in the process of creating tests for the change to limit the compression to `$tempdir/$user` when `$tempdir` contains a `$user` node.

Reply:
FWIW It doesn't actually interfere based on the changes I made to the existing tests and the function I created.